kolibri/core/assets/src/views/ContentRenderer/__tests__/DownloadButton.spec.js

```diff
@@ -0,0 +1,80 @@
+import Vue from 'vue';
+import { render, screen } from '@testing-library/vue';
+import VueRouter from 'vue-router';
+import useUser, { useUserMock } from 'kolibri.coreVue.composables.useUser';
+import DownloadButton from '../DownloadButton.vue';
+import { RENDERER_SUFFIX } from '../constants';
+import '@testing-library/jest-dom';
+
+jest.mock('kolibri.coreVue.composables.useUser');
+
+const getDownloadableFile = (isExercise = false) => {
+  const PRESET = isExercise ? 'exercise' : 'thumbnail';
+
+  // Register a component with the preset name so that the file is considered renderable
+  Vue.component(PRESET + RENDERER_SUFFIX, { template: '<div></div>' });
+
+  return {
+    preset: PRESET,
+    available: true,
+    file_size: 100,
+    storage_url: 'http://example.com/sample.png',
+    extension: 'png',
+    checksum: '1234567890',
+  };
+};
+
+// A helper function to render the component with the given props and some default mocks
+const renderComponent = props => {
+  useUser.mockImplementation(() =>
+    useUserMock({
+      isAppContext: props.isAppContext || false,
+    })
+  );
+
+  return render(DownloadButton, {
+    routes: new VueRouter(),
+    props: {
+      files: [],
+      nodeTitle: '',
+      ...props,
+    },
+  });
+};
+
+const SAVE_BUTTON_TEXT = 'Save to device';
+
+describe('DownloadButton', () => {
+  beforeEach(() => {
+    Vue.options.components = {};
+  });
+
+  test('does not render if isAppContext is true', () => {
+    renderComponent({ isAppContext: true });
+    expect(screen.queryByText(SAVE_BUTTON_TEXT)).not.toBeInTheDocument();
+  });
+
+  test('should not render if there are no downloadable files even if isAppContext is false', () => {
+    renderComponent({
```

Comment:
I think we need to find a better way of passing the useUserMock values, because reading just the testcase I could think that `isAppContext`is a prop of the component, but it isn't, so we can maybe wrap this value in a `useUserMock` object? This way the name suggest that we are passing this value to a mock of a composable. And we can end up in something like: 

```javascript
renderComponent({
  files: [],
  useUserMock: { isAppContext: false }
});
```

Let me know what do you think, as we can standarize this way of passing composables mock values.

Reply:
Indeed, this looks better! This provides a clear boundary between the props of the component, and all the other configuration that might be required by it, making it easier to reason about. Will update!